### PR TITLE
Resnet50 model 1 docs

### PR DESCRIPTION
**Pull request by rduvur**

### CODE REVIEW

File: content/analytics-team/resnet50_doc.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+### CNN Architecture
+
+#### ResNet50 Layers
+1. Stage 1
+   - Conv1 (7x7, 64 filters, stride = 2)
+   - Batch Norm
+   - ReLU
+   - Max Pooling
+
```
Comment (ahmostafa147) [inline content/analytics-team/resnet50_doc.md]: Need details about why this architecture.

File: content/analytics-team/resnet50_doc.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+1. Flatten Layer
+2. ReLU
+ - Added l2 regularization
+3. Dropout (0.3)
+4. ReLU
+5. Dropout (0.2)
+6. Sigmoid Activation
+
+## Training Configuration
+
```
Comment (ahmostafa147) [inline content/analytics-team/resnet50_doc.md:99]: need documentation of why did we choose these optimizers and loss and metrics.

File: content/analytics-team/resnet50_doc.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+# Compile model with Cosine Annealing scheduler
+model.compile(optimizer=Adam(learning_rate=cosine_decay_restarts),
+              loss='binary_crossentropy',
+              metrics=['accuracy'])
+
+# Callbacks
+early_stopping = EarlyStopping(monitor='val_loss', patience=10, restore_best_weights=True, verbose=1)
+model_checkpoint = ModelCheckpoint('best_model.keras', monitor='val_loss', save_best_only=True, verbose=1)
+```
+
```
Comment (ahmostafa147) [inline content/analytics-team/resnet50_doc.md:121]: What are some of the problems you encountered during the model tuning?

File: content/analytics-team/resnet50_doc.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+])
+
+# Implement Cosine Annealing
+initial_lr = 1e-4
+cosine_decay_restarts = CosineDecayRestarts(initial_learning_rate=initial_lr,
+                                            first_decay_steps=5000,
+                                            t_mul=2.0,  # Increases period after each restart
+                                            m_mul=0.8,  # Reduce max learning rate after each restart
+                                            alpha=1e-6)
+
```
Comment (ahmostafa147) [inline content/analytics-team/resnet50_doc.md]: no need to add code

File: content/analytics-team/resnet50_doc.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  - Loss
+  - Accuracy
+  - AUC
+  - Validation metrics
+
+## Dependencies
+- TensorFlow 2.x
+- Python 3.x
+- NumPy
+- Other requirements as listed in project's requirements.txt
```
Comment (ahmostafa147) [inline content/analytics-team/resnet50_doc.md]: there is no requirements.txt file in our project

**Review by ahmostafa147 — CHANGES_REQUESTED** [review]
Looks good as a starting point, however, need more details about other stuff.

File: content/analytics-team/resnet50_doc.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+title: "Analytics Team Documentation"
+date: 2025-02-16
+---
+
+# X-Ray Classification Model Documentation
+## Overview
+This document provides technical details about our ResNet50 Transfer Learning approach for extracting meaningful features from chest X-Rays and provide a binary classification to find lungs with pulmonary issues. 
+
+## Model Architecture
+
```
Comment (ahmostafa147) [inline content/analytics-team/resnet50_doc.md:12]: add a little details about where will these images come from. I mean by that is the images are gonna be preprocessed by some python functions in the preprocessing folder and will save the preprocessed version in the processed data, so just mention the output of which preprocessing function is gonna be the one used for this model.
